[sdformat] Avoid use of Environment.hh

We choose to opt-out the environment-variable-based include path tweaks
defined by the SDFormat specification. Our design intent is that include
files are always package:// URIs, never relative filenames.

--- src/SDF.cc
+++ src/SDF.cc
@@ -33,8 +33,6 @@
 #include "sdf/sdf_config.h"
 #include "EmbeddedSdf.hh"
 
-#include <gz/utils/Environment.hh>
-
 namespace sdf
 {
 inline namespace SDF_VERSION_NAMESPACE
@@ -132,7 +132,8 @@
   }
 
   std::string sdfPathEnv;
-  if(gz::utils::env("SDF_PATH", sdfPathEnv))
+  // Drake opts-out the environment-variable-based include path tweaks.
+  if(false)
   {
     std::vector<std::string> paths = sdf::split(sdfPathEnv, ":");
     for (std::vector<std::string>::iterator iter = paths.begin();
